Neos.Neos/Classes/Domain/Model/WorkspaceRoleAssignments.php

```diff
@@ -47,4 +96,19 @@ public function count(): int
     {
         return count($this->assignments);
     }
+
+    public function contains(WorkspaceRoleAssignment $assignment): bool
+    {
+        foreach ($this->assignments as $existingAssignment) {
+            if ($existingAssignment->equals($assignment)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public function add(WorkspaceRoleAssignment $assignment): self
```

Comment:
`withWorkspaceRoleAssignment` would better indicate the immutability here than `add`.